src/content/docs/cheerpj3/05-reference/Runtime-API.md

```diff
@@ -0,0 +1,444 @@
+var returnVal = cjCall(
+	"com.my.Java.package.myClass",
+	"method",
+	argument1,
+	argument2,
+	argument3,
+);
+```
+
+Example 2:
+
+```js
+/* Equivalent Java code: int returnVal = object.method(argument1, argument2, argument3); */
+var returnVal = cjCall(object, "method", argument1, argument2, argument3);
+```
+
+### cjNew
+
+`cjNew(classNameOrInvoker, ...)`
+
+Calls Java constructors from JavaScript.
+
+Example:
+
+```js
+/* Equivalent Java code: myClass object = com.my.Java.package.myClass(argument1)  */
+var object = cjNew("com.my.Java.package.myClass", argument1);
+```
+
+### cjResolveCall / cjResolveNew
+
+`cjResolveCall(className, methodName, types)`
+
+`cjResolveNew(className, types)`
+
+Using `cjCall/cjNew` is convenient, but under the hood Java reflection APIs are used, which may have a significant performance impact if used heavily. If you plan to use `cjCall/cjNew` many times it is convenient to go through reflection APIs only once by using the `cjResolveCall/cjResolveNew` API.
+
+Examples:
+
+```js
+var promise1 = cjResolveCall("com.something.ClassName", "methodName", [
+	"java.lang.String",
+	"int",
+	"double",
+]);
+var promise2 = cjResolveNew("com.something.ClassName", [
+	"java.lang.String",
+	"int",
+	"double",
+]);
+```
+
+`cjResolveCall` supports both instance and static methods of classes. The third parameter (for both APIs) is an array of Java types and it is only required if the methodName is not unique in the class (i.e. it is overloaded). The third parameter can be omitted (be null) if the method name (or constructor) is unique. For example:
+
+```js
+var promise = cjResolveCall(
+	"com.something.ClassName",
+	"uniqueMethodName",
+	null,
+);
+```
+
+`cjResolveCall/cjResolveNew` are async, like most of CheerpJ's APIs. To get the actual result you can either use .then() or the async/await functionality of JS. For example:
+
+```js
+var resolvedMethod = await cjResolveCall("com.something.ClassName", "uniqueMethodName", null);
+cjResolveCall("com.something.ClassName", "uniqueMethodName", null).then(function(resolvedMethod) { ... });
+```
+
+The returned value is an opaque handle to the desired method (or constructor), which can now be called an arbitrary number of times without going through Java reflection.
+
+```js
+cjCall(resolvedMethod, arg1, arg2, arg3);
+cjNew(resolvedConstructor, arg1, arg2, arg3);
+```
+
+Alternatively resolvedMethod can also be used _directly as a function_, for example:
+
+```js
+resolvedMethod(arg1, arg2, arg3);
+```
+
+Please note that this convenient form can unfortunately only be used on the main thread, not on Workers. For more information see [WebWorker API](/cheerpj3/reference/WebWorker-API)
+
+## Data conversion
+
+### cjStringJavaToJs
+
+`cjStringJavaToJs(javaString)`
+
+```js
+var jsString = cjStringJavaToJs(javaString);
+```
+
+This converts a Java string into a JavaScript string. This operation implies a copy.
+
+### cjStringJsToJava
```

Comment:
This APIs do not exist anymore, implicit conversions are applied when using library mode and JS natives